File: wk7/java/aliceBobAndChocolate.java
// problem link: https://codeforces.com/problemset/problem/6/C
// solution link: https://codeforces.com/contest/6/submission/285500257

import java.io.*;
import java.math.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader r = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(r.readLine());
        int[] a = Arrays.stream(r.readLine().split(" ")).mapToInt(Integer::parseInt).toArray();
        int l = 0, h = n - 1, c = 0;
        while(l <= h) {
            if(c <= 0)
                c += a[l++];
            else
                c -= a[h--];
        }
        System.out.println(l + " " + (a.length - l));
    }
}

File: wk7/cpp/aliceBobAndChocolate.cpp
// problem link: https://codeforces.com/problemset/problem/6/C
// solution link: https://codeforces.com/contest/6/submission/291813118

#include <bits/stdc++.h>
using namespace std;

int main()
{
    ios::sync_with_stdio(false); cin.tie(0); cout.tie(0);
    
    int n, x;
    cin >> n;
    vector<int> a;
    for(int i = 0; i < n; i++) {
        cin >> x;
        a.push_back(x);
    }
    int l = 0, h = n - 1, c = 0;
    while(l <= h) {
        if(c <= 0)
            c += a[l++];
        else
            c -= a[h--];
    }
    cout << l << " " << (a.size() - l);

    return 0;
}